itou/approvals/export.py

```diff
@@ -73,11 +49,25 @@ def export_approvals(export_format="file"):
     ws.title = "Export PASS SIAE " + current_dt.strftime(DATE_FMT)
 
     data = [FIELDS]
-
-    for approval in Approval.objects.valid():
-        new_line = _approval_line(approval)
-        if new_line:
-            data.append(_approval_line(approval))
+    approvals = Approval.objects.all().select_related("user").prefetch_related("jobapplication_set__to_siae")
```

Comment:
Quelques remarques ici :

- on exporte tous les PASS IAE depuis le début même ceux expirés d'où l'utilisation du `all`
- on utilise [`select-related`](https://docs.djangoproject.com/en/3.0/ref/models/querysets/#select-related) et [`prefetch-related`](https://docs.djangoproject.com/en/3.0/ref/models/querysets/#prefetch-related) pour faire les jointures SQL avec l'ORM de Django et éviter le problème des requêtes "N+1" des ORMs

Reply:
TIL la différence entre select-related et prefetch-related :

_both are designed to stop the deluge of database queries that is caused by accessing related objects, but the strategy is quite different.

select_related works by creating an SQL join and including the fields of the related object in the SELECT statement. For this reason, select_related gets the related objects in the same database query. However, to avoid the much larger result set that would result from joining across a ‘many’ relationship, select_related is limited to single-valued relationships - foreign key and one-to-one.

prefetch_related, on the other hand, does a separate lookup for each relationship, and does the ‘joining’ in Python. This allows it to prefetch many-to-many and many-to-one objects, which cannot be done using select_related, in addition to the foreign key and one-to-one relationships that are supported by select_related._